lib/api/normalize.js

```diff
@@ -0,0 +1,307 @@
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+
+        const changedBlocks = blockMap
+            .filter(block => block.getDepth() > maxListNesting)
+            .map(block => block.set('depth', maxListNesting));
+
+        if (changedBlocks.size !== 0) {
+            return EditorState.set(editorState, {
+                currentContent: content.merge({
+                    blockMap: blockMap.merge(changedBlocks),
+                }),
+            });
+        }
+
+        return editorState;
+    },
+
+    /**
+     * Resets all blocks that use unavailable types to unstyled.
+     */
+    resetBlockType(editorState, enabledTypes) {
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+
+        const changedBlocks = blockMap
+            .filter(block => !enabledTypes.includes(block.getType()))
+            .map(block => block.set('type', BLOCK_TYPE.UNSTYLED));
+
+        if (changedBlocks.size !== 0) {
+            return EditorState.set(editorState, {
+                currentContent: content.merge({
+                    blockMap: blockMap.merge(changedBlocks),
+                }),
+            });
+        }
+
+        return editorState;
+    },
+
+    /**
+     * Removes all styles that use unavailable types.
+     */
+    filterInlineStyle(editorState, enabledTypes) {
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+
+        const blocks = blockMap.map(block => {
+            let altered = false;
+
+            const chars = block.getCharacterList().map(char => {
+                let newChar = char;
+
+                char
+                    .getStyle()
+                    .filter(type => !enabledTypes.includes(type))
+                    .forEach(type => {
+                        altered = true;
+                        newChar = CharacterMetadata.removeStyle(newChar, type);
+                    });
+
+                return newChar;
+            });
+
+            return altered ? block.set('characterList', chars) : block;
+        });
+
+        return EditorState.set(editorState, {
+            currentContent: content.merge({
+                blockMap: blockMap.merge(blocks),
+            }),
+        });
+    },
+
+    /**
+     * Resets atomic blocks to unstyled based on which entity types are enabled,
+     * and also normalises block text to a single "space" character.
+     */
+    resetAtomicBlocks(editorState, enabledTypes) {
+        const content = editorState.getCurrentContent();
+        const blockMap = content.getBlockMap();
+        let blocks = blockMap;
+
+        const normalisedBlocks = blocks
+            .filter(
+                block =>
+                    block.getType() === BLOCK_TYPE.ATOMIC &&
+                    (block.getText() !== ' ' ||
+                        block.getInlineStyleAt(0).size !== 0),
```

Comment:
Normalises the block if its text isn't right, or if it has inline styles on it. Happens with pasting from Google Docs.